### ISSUE TITLE
Request for vscode plugin feature to pull all updates (IDFGH-11260)

### ISSUE THREAD

**Issue by baylf2000** [espressif/esp-idf#12418]
### Is your feature request related to a problem?

It can be difficult and complicated for beginners to keep all the Espressif repositories fully updated including submodules. I wasted many hours today trying to trace the origins of a failing build only to find that some updates had occurred that were causing the build errors. `git pull` had pulled part of the updates but not the submodule updates, which then caused the build to fail.

An easier solution, especially for beginners, would be handy.

### Describe the solution you'd like.

An automated way to keep esp-idf and esp-matter completely updated would be great. Or even a manual process like an "ESP-IDF:" command in the vscode plugin that pulled all the latest updates from github, including other related packages like esp-matter.

**Comment by chshu** [comment]
@baylf2000 Yes, after `git pull`, you need to update submodules to ensure every submodule is up to date.

Regarding the vscode plugin, I'm not if it's flexible enough for all the SDKs, since different customer may need different repos.

BTW, you can try out the esp-matter component: https://docs.espressif.com/projects/esp-matter/en/latest/esp32/developing.html#esp-matter-component-experimental, so you don't need to download esp-matter repo.
